watchers/disruptions_watchers_manager_test.go

```diff
@@ -0,0 +1,467 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers_test
+
+import (
+	"fmt"
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/mocks"
+	"github.com/DataDog/chaos-controller/watchers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+)
+
+var _ = Describe("Disruptions watchers manager", func() {
+	var (
+		chaosPodWatcherMock                                              *watchers.WatcherMock
+		cacheMock                                                        *CacheMock
+		disruption                                                       *chaosv1beta1.Disruption
+		disruptionsWatchersManager                                       watchers.DisruptionsWatchersManager
+		disruptionTargetWatcherMock                                      *watchers.WatcherMock
+		err                                                              error
+		expectedDisruptionTargetWatcherName, expectedChaosPodWatcherName string
+		watchersManagerMock                                              *watchers.ManagerMock
+		watcherFactoryMock                                               *watchers.FactoryMock
+		readerMock                                                       *mocks.ReaderMock
+	)
+
+	twoDisruptions := []*chaosv1beta1.Disruption{
```

Comment:
FYI if inside your setup or assertion nodes you modify those at any time you should NOT assign them here but in a `BeforeEach` node.

To avoid confusion and potential later usage and bug, you may still want to move this into a `BeforeEach` now so later contributors won't have to fix this.

Only case it's OK to declare + assign is for const IMO.